From 169acb91269f9c4a9380115ca2f8ab0aca364adb Mon Sep 17 00:00:00 2001
From: keyur-mpc <keyur@mpconsultants.org>
Date: Tue, 28 Nov 2023 19:20:40 +0000
Subject: [PATCH 14/14] Rev02 DTS changes 1. sdcard cd line changes 2. added 3
 status led gpios 3. bbsensor onewire update 4. removed uart4 from dwx 5.
 Added extra adc pin configuration on gpio1_io4 6. Added 4 extra gpios

---
 ...ll-var-som-concerto-board-emmc-sd-card.dts | 23 ------
 ...x6ull-var-som-concerto-board-emmc-wifi.dts | 72 ++++++++++++++-----
 2 files changed, 53 insertions(+), 42 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-sd-card.dts b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-sd-card.dts
index 24562c37fed3..dcf036d532dc 100644
--- a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-sd-card.dts
+++ b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-sd-card.dts
@@ -19,27 +19,4 @@
 / {
 	model = "Variscite VAR-SOM-6UL with i.MX6ULL, eMMC & SD card support on Concerto-Board";
 	compatible = "variscite,imx6ul-var-som", "fsl,imx6ull";
-
-	/* Deleting this node to avoid conflict with GPIO1_IO04 for SD card CD pin */
-	/delete-node/ pinctrl_tsc;
-	/delete-node/ tsc;
-};
-
-
-&iomuxc {
-	/* Using Pin 189 */
-	pinctrl_usdhc1_gpio: usdhc1_gpiogrp {
-		fsl,pins = <
-			MX6UL_PAD_GPIO1_IO04__GPIO1_IO04	0x1b0b1 /* CD */
-		>;
-	};
-};
-
-/* Disabled to avoid conflicts with GPIO1_IO04 */
-&tsc {
-    status = "disabled";
-};
-
-&usdhc1 {
-	cd-gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
 };
diff --git a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
index a94f3bc8bf6d..ca358e1b224d 100644
--- a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
+++ b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
@@ -40,16 +40,23 @@
     /delete-node/ pinctrl_pwm4;
     /delete-node/ pwm4;
 
+    /delete-node/ pinctrl_tsc;
     /delete-node/ tsc;
 
     /delete-node/ backlight;
 
+    /delete-node/ pinctrl_gpio_keys;
+    /delete-node/ pinctrl_gpio_keys_snvs;
+    /delete-node/ back;
+    /delete-node/ wakeup;
+    /delete-node/ keys;
+
     run-control {
         compatible = "gpio-leds";
         pinctrl-names = "default";
         pinctrl-0 = <&pinctrl_java_gpio>, <&pinctrl_java_gpio_snvs>;
 		status = "okay";
-        dwx_ls {
+        ds18b20_ls {
             gpios = <&gpio3 9 GPIO_ACTIVE_HIGH>;
             default-state = "off";
         };
@@ -97,11 +104,33 @@
             gpios = <&gpio4 23 GPIO_ACTIVE_HIGH>;
             default-state = "off";
         };
-        dwx_rst {
-            gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+        led_r {
+            gpios = <&gpio3 6 GPIO_ACTIVE_HIGH>;
+            default-state = "off";
+        };
+        led_g {
+            gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
+            default-state = "off";
+        };
+        led_b {
+            gpios = <&gpio4 14 GPIO_ACTIVE_HIGH>;
             default-state = "off";
         };
     };
+
+    extra-gpios {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_java_extra_gpio>;
+		status = "okay";
+    };
+
+    onewire {
+        compatible = "w1-gpio";
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_w1>;
+        gpios = <&gpio1 28 GPIO_ACTIVE_HIGH>;
+    };
+
 };
 
 /* MPC changes*/
@@ -127,13 +156,6 @@
         >;
     };
 
-    pinctrl_uart4: uart4grp {
-        fsl,pins = <
-            MX6UL_PAD_UART4_TX_DATA__UART4_DCE_TX	0x1b0b1
-            MX6UL_PAD_UART4_RX_DATA__UART4_DCE_RX	0x1b0b1
-        >;
-    };
-
     pinctrl_uart6: uart6grp {
         fsl,pins = <
             MX6UL_PAD_CSI_MCLK__UART6_DCE_TX	0x1b0b1
@@ -146,6 +168,7 @@
     pinctrl_adc1: adc1grp {
 		fsl,pins = <
 			MX6UL_PAD_GPIO1_IO01__GPIO1_IO01 0xb0
+            MX6UL_PAD_GPIO1_IO04__GPIO1_IO04 0xb0
 		>;
 	};
 
@@ -160,7 +183,19 @@
 			MX6UL_PAD_GPIO1_IO02__GPIO1_IO02 0x1b0b0
 			MX6UL_PAD_LCD_DATA02__GPIO3_IO07 0x1b0b0
 			MX6UL_PAD_CSI_DATA02__GPIO4_IO23 0x1b0b0
+            MX6UL_PAD_LCD_DATA01__GPIO3_IO06 0x1b0b0
+            MX6UL_PAD_LCD_DATA00__GPIO3_IO05 0x1b0b0
+            MX6UL_PAD_NAND_CE1_B__GPIO4_IO14 0x1b0b0
+		>;
+	};
+
+    pinctrl_java_extra_gpio: javaextragpiogrp {
+		fsl,pins = <
 			MX6UL_PAD_GPIO1_IO05__GPIO1_IO05 0x1b0b0
+            MX6UL_PAD_CSI_DATA06__GPIO4_IO27 0x1b0b0
+            MX6UL_PAD_CSI_DATA04__GPIO4_IO25 0x1b0b0
+            MX6UL_PAD_CSI_DATA05__GPIO4_IO26 0x1b0b0
+            MX6UL_PAD_CSI_DATA07__GPIO4_IO28 0x1b0b0
 		>;
 	};
 
@@ -169,6 +204,13 @@
             MX6UL_PAD_UART5_RX_DATA__GPIO1_IO31	0x1b0b0 /* fec2 reset */
         >;
     };
+
+    pinctrl_w1: w1grp {
+        fsl,pins = <
+            MX6UL_PAD_UART4_TX_DATA__GPIO1_IO28   0x0c0b0
+       >;
+    };
+
 };
 
 /* SVNS pad needs to be configured in iomuxc_snvs */
@@ -205,14 +247,6 @@
 	status = "okay";
 };
 
-/* ttymxc3 UART */
-&uart4 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart4>;
-	/*fsl,uart-has-rtscts;*/
-	status = "okay";
-};
-
 /* ttymxc5 UART */
 &uart6 {
 	pinctrl-names = "default";
@@ -242,7 +276,7 @@
     status = "disabled";
 };
 
-/* Disabled to avoid conflicts with GPIO1_IO02 */
+/* Disabling this node to avoid conflict with GPIO1_IO04 for GPIO1_IO04 and GPIO1_IO02 pin */
 &tsc {
     status = "disabled";
 };
-- 
2.25.1

